common/environment/setup/install.sh

```diff
@@ -258,3 +258,34 @@ _vmkdir() {
 		install -dm${mode} ${_destdir}/${dir}
 	fi
 }
+
+_vcompletion() {
+	local cmd="$1" shell="$2" file="$3"
+	local _bash_completion_dir=usr/share/bash-completion/completions/
+	local _fish_completion_dir=usr/share/fish/vendor_completions.d/
+	local _zsh_completion_dir=usr/share/zsh/site-functions/
+
+	if [ -z "$DESTDIR" ]; then
+		msg_red "$pkgver: vcompletion: DESTDIR unset, can't continue...\n"
+		return 1
+	fi
+
+	if [ -z "$cmd" -o -z "$shell" ]; then
+		msg_red "$pkgver: vcompletion: command or shell argument unset.\n"
+		return 1
+	fi
+
+	: "${file:=${cmd}.${shell}}"
```

Comment:
Default value is specific for completions generated by these two packages.  Currently most of completions are installed from some subdir, so maybe let's make `file` first argument, like other vinstall based functions.

Reply:
Agree, then we could guess `shell` from filename, and `cmd` from `pkgname`
```sh
file="$1"
cmd="${2:-$pkgname}"
shell="${3:-${file##*.}}"
```

Should we guess `cmd` part from `file` as well?

Reply:
> cmd from pkgname

Sure, but then check for existing target file is needed to prevent overwriting by consecutive calls.

> shell from filename
> shell="${3:-${file##*.}}"

It would need to be more like `*.fish => fish, *bash* => bash, _* => zsh, else error` to fit real world, which is too unpredictable and matching nothing too often imo.

Reply:
I left the order as `file shell command`, with only command being optional, since it can be taken from `pkgname`. As @Chocimier said, guessing the shell from the filename will be a bit of a pain for little gain.